server/server.go

```diff
@@ -203,11 +235,55 @@ func isHopByHopHeader(name string) bool {
 	}
 }
 
-func (p *proxy) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	if backendID := r.Header.Get(utils.HeaderBackendID); backendID != "" {
-		p.handleAgentRequest(w, r, backendID)
-		return
+func websocketShimResponseHandlerOpen(resp *http.Response, ws *wsSessionHelper) error {
+	p, err := io.ReadAll(resp.Body)
```

Comment:
This method doesn't handle the case that the status in the open response is not `http.StatusOK`.

Additionally, the corresponding logic in the other response handler methods is duplicated.

I think we should move that to the `handleFrontendRequest` method so that we avoid duplicating it.

I'll leave a comment below for my suggestion on how to do that.